### PR TITLE
III-4572 Handle empty time spans on event copy

### PR DESCRIPTION
**Pull request by LucWollants**
### Fixed
- Handled empty time spans also on event copy

---
Ticket: https://jira.uitdatabank.be/browse/III-4572

### CODE REVIEW

File: src/Http/Offer/LegacyTimeSpansParser.php [hunk 1]
```diff
@@ -16,22 +16,25 @@ public function parse(stdClass $data): StdClass
...
-                    // Rename end to endDate
-                    if ($timeSpan instanceof stdClass && isset($timeSpan->end)) {
-                        $timeSpan->endDate = $this->formatDateTime($timeSpan->end);
-                        unset($timeSpan->end);
-                    }
-                    return $timeSpan;
-                },
-                $data->timeSpans
-            );
+            if (count($data->timeSpans) > 0) {
```
Comment (bertramakers) [inline src/Http/Offer/LegacyTimeSpansParser.php:19]: Nit: Can this `if` not be part of the `if` above it?

```
if (isset($data->timeSpans) && is_array($data->timeSpans) && !isset($data->subEvent) && count($data->timeSpans) > 0))
```

**Review by bertramakers — APPROVED** [review]

File: src/Http/Offer/LegacyTimeSpansParser.php (hunk 1 above)
Comment (bertramakers) [reply src/Http/Offer/LegacyTimeSpansParser.php:19]: Looks good otherwise though!

Comment (LucWollants) [reply src/Http/Offer/LegacyTimeSpansParser.php:19]: I always wanted to do the `unset($data->timeSpans);`

Comment (bertramakers) [reply src/Http/Offer/LegacyTimeSpansParser.php:19]: Ah I missed that! Good call.

**Pull request merged**